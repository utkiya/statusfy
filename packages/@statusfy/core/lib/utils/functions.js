const { fse, grayMatter, chalk, path } = require("@statusfy/common");

const getIncidentsFromProject = async contentDir => {
  const files = await fse.readdir(contentDir);
  const incidentsList = [];

  for (let i = 0; i < files.length; i++) {
    const f = path.resolve(contentDir, files[i]);
    const ext = path.extname(f);
    const fileName = path.basename(f);

    if (ext === ".md") {
      const fileContent = await fse.readFile(f);
      const { data } = grayMatter.parse(fileContent);

      incidentsList.push({
        value: {
          name: fileName,
          path: f
        },
        name: `${fileName} > ${chalk.yellow(data.title)} (${chalk.green(
          new Date(data.date).toUTCString()
        )})`
      });
    }
  }

  return incidentsList;
};

const generateIncident = (data, content, format) => {
  let matterContent = grayMatter.stringify(content, data, format);

  if (["json", "toml"].includes(format)) {
    matterContent = matterContent.replace(/^---/, `---${format}`);
  }

  return matterContent;
};

module.exports = {
  getIncidentsFromProject,
  generateIncident
};
