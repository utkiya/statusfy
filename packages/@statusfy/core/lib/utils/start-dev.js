const portfinder = require("portfinder");
const bs = require("browser-sync").create();

const { logger, path } = require("@statusfy/common");
const createServer = require("../../server");
const generateConfig = require("../config/generate");

/* eslint-disable require-await */
const start = async () => {
  process.env.NODE_ENV = "development";
  const sourceDir = process.env.STATUSFY_SOURCE_DIR;
  const cliOptions = JSON.parse(process.env.STATUSFY_CLI_OPTIONS);

  // Generate configuration
  const { nuxtConfig, siteConfig } = generateConfig(sourceDir, cliOptions);
  nuxtConfig.mode = cliOptions.ssr ? "universal" : "spa";
  delete nuxtConfig.build.analyze;

  portfinder
    .getPortPromise()
    .then(async port => {
      const serverPort = nuxtConfig.server.port;

      // Update server port
      nuxtConfig.server.port = port;

      // Initialize the server
      await createServer(
        siteConfig,
        nuxtConfig,
        nuxtConfig.server.host,
        nuxtConfig.server.port
      );

      // Create borwser-sync server
      bs.init({
        port: serverPort,
        proxy: `${nuxtConfig.server.host}:${nuxtConfig.server.port}`,
        open: false
      });

      bs.watch(path.join(sourceDir, siteConfig.content.dir, "**/*.md")).on(
        "change",
        bs.reload
      );
    })
    .catch(err => {
      logger.error(err);
    });
};

start();
