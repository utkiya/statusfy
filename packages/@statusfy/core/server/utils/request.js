module.exports = req => {
  const language = req.params.lang || req.app.get("language");
  const page = req.query.page || req.params.lang || 1;

  return {
    language,
    page,
    siteConfig: req.app.get("siteConfig")
  };
};
