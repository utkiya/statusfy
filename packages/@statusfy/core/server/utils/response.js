module.exports = (res, lang) => ({
  json(data) {
    res.setHeader("Content-Language", lang);
    res.json(data);
  },
  notFound(message = "Not Found") {
    res.setHeader("Cache-Control", "no-cache");
    res.status(404).json({
      message
    });
  }
});
