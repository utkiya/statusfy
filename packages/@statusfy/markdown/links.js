module.exports = md => {
  // Remember old renderer, if overridden, or proxy to default renderer
  const defaultRender = md.renderer.rules.link_open || ((tokens, idx, options, env, self) => self.renderToken(tokens, idx, options))

  md.renderer.rules.link_open = (tokens, idx, options, env, self) => {
    const token = tokens[idx]
    const hIndex = token.attrIndex('href')
    const url = token.attrs[hIndex][1]
    const isInternalLink = (url.startsWith('/') && !url.startsWith('//') ) || url.startsWith('./') || url.startsWith('../')

    if (!isInternalLink) {
      token.attrSet('target', '_blank')
      token.attrSet('rel', 'noopener noreferrer')
      token.attrSet('class', 'external')
    }

    // pass token to default renderer.
    return defaultRender(tokens, idx, options, env, self)
  }
}
