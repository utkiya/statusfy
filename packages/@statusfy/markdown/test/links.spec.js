import { md } from './utils'
import links from '@/markdown/links.js'

const mdL = md().use(links)

const internalLinkAsserts = {
  '/': '/',
  '/foo/': '/foo/',
  '/foo/#hash': '/foo/#hash'
}

const externalLinks = [
  '[Statusfy](https://statusfy.co)',
  '[Statusfy](http://statusfy.co)',
  '[Statusfy Docs](http://docs.statusfy.co)'
]

describe('links', () => {
  test('should render internal links correctly', () => {
    for (const before in internalLinkAsserts) {
      const input = `[${before}](${before})`
      const output = mdL.render(input)

      expect(output).toMatchSnapshot()
    }
  })

  test('should render external links correctly', () => {
    for (const link of externalLinks) {
      const output = mdL.render(link)

      expect(output).toMatchSnapshot()
    }
  })
})
