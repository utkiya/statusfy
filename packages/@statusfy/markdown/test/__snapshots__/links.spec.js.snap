// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`links should render external links correctly 1`] = `<p><a href="https://statusfy.co" target="_blank" rel="noopener noreferrer" class="external">Statusfy</a></p>`;

exports[`links should render external links correctly 2`] = `<p><a href="http://statusfy.co" target="_blank" rel="noopener noreferrer" class="external">Statusfy</a></p>`;

exports[`links should render external links correctly 3`] = `<p><a href="http://docs.statusfy.co" target="_blank" rel="noopener noreferrer" class="external">Statusfy Docs</a></p>`;

exports[`links should render internal links correctly 1`] = `<p><a href="/">/</a></p>`;

exports[`links should render internal links correctly 2`] = `<p><a href="/foo/">/foo/</a></p>`;

exports[`links should render internal links correctly 3`] = `<p><a href="/foo/#hash">/foo/#hash</a></p>`;
